api/clients/v2/relay_payload_retriever.go

```diff
@@ -0,0 +1,289 @@
+package clients
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	"github.com/Layr-Labs/eigenda/common/geth"
+	verifiercontract "github.com/Layr-Labs/eigenda/contracts/bindings/EigenDACertVerifier"
+	core "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/consensys/gnark-crypto/ecc/bn254"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// RelayPayloadRetriever provides the ability to get payloads from the relay subsystem.
+//
+// This struct is goroutine safe.
+type RelayPayloadRetriever struct {
```

Comment:
I personally MSB order (see https://github.com/tigerbeetle/tigerbeetle/blob/main/docs/TIGER_STYLE.md#naming-things), so that the implementations of PayloadRetriever would be:
- PayloadRetrieverRelay
- PayloadRetrieverS3
- PayloadRetrieverNodes

This makes it much easier to find them using simple ctrl-f, or just searching for `PayloadRetriever`

But not going to push super hard on this one if you don't like it

Reply:
I'm torn on this one. In general I agree, but the name `PayloadRetrieverRelay` implies that this thing *IS* a relay of some kind. I think the potential confusion there outweighs the `ctrl-f` benefit

Reply:
> PayloadRetrieverS3

Do we have a payload retriever for S3? Is this purely an internal use tool? The general public shouldn't have direct access to S3, except as mitigated through the relays.

Reply:
My personal preference on word ordering would be `RelayPayloadRetriever`.

Reply:
> Do we have a payload retriever for S3

This isn't referring to any eigen DA controlled S3. Rather, some rollups might have their own home-rolled storage backends.

Reply:
I am outvoted. Resolving.

Regarding "PayloadRetrieverRelay implies that this thing IS a Relay". That just requires some brain restructuring, the same way reverse polish notation does. But I'm not going to fight this fight. Agree that it might make more sense in other scenarios that are less syntactically overloaded.